util/failpoint.go

```diff
@@ -33,13 +33,34 @@
 package util
 
 import (
+	"errors"
+
 	"github.com/pingcap/failpoint"
+	"go.uber.org/atomic"
 )
 
 const failpointPrefix = "tikvclient/"
 
+var (
+	failpointsEnabled atomic.Bool
```

Comment:
Can we use a global flag and setup to `true` from `go` cli when testing? `atomic.Bool` seems another significant overhead.

Reply:
That is a very good point. The good side of using `bool` is performance, while the down side is that we can only decide whether to enable failpoints before starting using client-go. It basically means we need to set up it in `init()` or `main` or there could be race conditions.

The situation is that the `tidb-server` can provide services through HTTP and can be controlled in a dynamic way (of course, the tidb-server should be compiled with `failpoint-ctl enable` for code generation).

I think a better option could be using `bool + atomic.Bool` way. `bool` should be set at initial phase, `atomic.Bool` can be set dynamically. The failpoints only work when both values are `true`.

Looking forward to your feedback :)

Reply:
cc @tiancaiamao

I'd like to understand the previous code generate way first. In that way how we decide whether or not a failpoint is enabled?

Reply:
Previously, if build without failpoint, there is no cost at all. If build after failpoint code generation, the cost is about the same as client-go before this PR.

Reply:
@disksing yep. I think it may be a bit overkill we have two options because one option keeps the same as previous flexibility. But I'm ok with two options for now since we can deprecated the inner `atomic.Bool` if it is proven to be redundant. Also, users enable failpoint should be already aware of the cost.

Reply:
After careful consideration, I think what you said makes sense. I decided to use a single bool variable for now, and it can't be too late to add atomic if there is a subsequent need.